lib/addons/xray-adot-addon/index.ts

```diff
@@ -0,0 +1,79 @@
+import { KubernetesManifest } from "aws-cdk-lib/aws-eks";
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { dependable, loadYaml, readYamlDocument } from "../../utils";
+import { AdotCollectorAddOn } from "../adot";
+import { Construct } from 'constructs';
+
+/**
+ * Configuration options for add-on.
+ */
+ export interface XrayAdotAddOnProps {
+    /**
+     * Modes supported : `deployment`, `daemonset`, `statefulSet`, and `sidecar`
+     * @default deployment
+     */
+    deploymentMode?: XrayDeploymentMode;
+    /**
+     * Namespace to deploy the ADOT Collector for XRay.
+     * @default default
+     */
+    namepace?: string;
+}
+
+export const enum XrayDeploymentMode {
+    DEPLOYMENT = 'deployment',
+    DAEMONSET = 'daemonset',
+    STATEFULSET = 'statefulset',
+    SIDECAR = 'sidecar'
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps = {
+    deploymentMode: XrayDeploymentMode.DEPLOYMENT,
+    namespace: 'default'
+};
+
+/**
+ * Implementation of XRAY ADOT add-on for EKS Blueprints. Installs ADOT Collector.
+ */
+export class XrayAdotAddOn implements ClusterAddOn {
+
+    readonly xrayAddOnProps: XrayAdotAddOnProps;
+
+    constructor(props?: XrayAdotAddOnProps) {
+        this.xrayAddOnProps = { ...defaultProps, ...props };
+    }
+
+    @dependable(AdotCollectorAddOn.name)
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let finalDeploymentMode: string;
+        let finalNamespace: string;
+        let doc: string;
+        
+        finalDeploymentMode = defaultProps.deploymentMode;
+        if (this.xrayAddOnProps.deploymentMode) {
+            finalDeploymentMode = this.xrayAddOnProps.deploymentMode;
+        }
+
+        finalNamespace = defaultProps.namespace;
+        if (this.xrayAddOnProps.namepace) {
+            finalNamespace = this.xrayAddOnProps.namepace;
+        }
+
+        // Applying manifest for configuring ADOT Collector for Xray.
+        doc = readYamlDocument(__dirname + '/collector-config-xray.ytpl');
```

Comment:
There is no helm chart for this? How will know that there is an update for this? I am unclear on the whole xray approach, it seems the product management is targeting POC level features as opposed to solid operational experience. 

Now to decouple ourselves from the YAML upgrades, we need to come up with a mechanism to enable customers to apply those without waiting on the next release of blueprints. Let's talk.